### PR TITLE
Limit the collections that the iast visitor can handle

### PR DESCRIPTION
**Pull request by manuel-alvarez-alvarez**
# What Does This Do
Limits the types of collections that IAST object visitor can handle to:

- Java collections
- Apache commons collections
- Guava collections
- Protobuf collections

# Motivation
When an hibernate persistent collection is added to the session scope, we will try to visit the different elements in order to detect trust boundary violation vulnerabilities and we might trigger issues further down the line due to lazy loading in ORMs.

# Additional Notes

# Contributor Checklist

- [ ] Format the title [according the contribution guidelines](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#title-format)
- [ ] Assign the `type:` and (`comp:` or `inst:`) labels in addition to [any usefull labels](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#labels)
- [ ] Squash your commits prior merging or merge using GitHub's [Squash and merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits)
- [ ] Don't use `close`, `fix` or any [linking keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) when referencing an issue.  
  Use `solves` instead, and assign the PR [milestone](https://github.com/DataDog/dd-trace-java/milestones) to the issue
- [ ] Update the [public documentation](https://docs.datadoghq.com/tracing/trace_collection/library_config/java/) in case of new configuration flag or behavior

Jira ticket: [SCRS-1113]

[SCRS-1113]: https://datadoghq.atlassian.net/browse/SCRS-1113?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: dd-java-agent/agent-iast/src/test/groovy/com/datadog/iast/GrpcRequestMessageHandlerTest.groovy [hunk 1]
```diff
@@ -66,7 +66,6 @@ class GrpcRequestMessageHandlerTest extends IastModuleImplTestBase {
     given:
     final visitor = Mock(ObjectVisitor.Visitor) {
       visit(_ as String, _ as Object) >> {
-        println 'feo'
```
Comment (manuel-alvarez-alvarez) [inline dd-java-agent/agent-iast/src/test/groovy/com/datadog/iast/GrpcRequestMessageHandlerTest.groovy:69]: Unrelated print of ferrous oxide :smile:

**Review by Mariovido — APPROVED** [review]
LGTM

**Review by amarziali — APPROVED** [review]

**Review by smola — APPROVED** [review]

File: dd-java-agent/agent-iast/src/test/groovy/com/datadog/iast/util/ObjectVisitorTest.groovy [hunk 2]
```diff
@@ -74,6 +70,29 @@ class ObjectVisitorTest extends Specification {
     0 * _
   }
 
+  void 'test visiting ignored collection'() {
```
Comment (vandonr) [inline dd-java-agent/agent-iast/src/test/groovy/com/datadog/iast/util/ObjectVisitorTest.groovy:73]: it's not clear to me what we're asserting in this test... I understand that we're not supposed to access the list, so shouldn't we throw an exception in the `get` for instance ? Or count accesses and make sure it's ==0 ?

Comment (manuel-alvarez-alvarez) [reply dd-java-agent/agent-iast/src/test/groovy/com/datadog/iast/util/ObjectVisitorTest.groovy:73]: Yes, we could throw an exception but in the end we are asserting that there are no mock invocations of the visitor with `0 * _` which is equivalent.

Comment (vandonr) [reply dd-java-agent/agent-iast/src/test/groovy/com/datadog/iast/util/ObjectVisitorTest.groovy:73]: OK thanks I wasn't familiar with that notation

**Pull request merged**